src/elements/content-sidebar/activity-feed/activity-feed/ActiveState.js

```diff
@@ -296,42 +300,65 @@ const ActiveState = ({
                                 isFocused={isFocused}
                                 ref={refValue}
                             >
-                                <ActivityThread
-                                    data-testid="activity-thread"
-                                    currentUser={currentUser}
-                                    getAvatarUrl={getAvatarUrl}
-                                    getMentionWithQuery={getMentionWithQuery}
-                                    getUserProfileUrl={getUserProfileUrl}
-                                    hasNewThreadedReplies={hasNewThreadedReplies}
-                                    hasReplies={hasReplies}
-                                    isPending={item.isPending}
-                                    isRepliesLoading={item.isRepliesLoading}
-                                    mentionSelectorContacts={mentionSelectorContacts}
-                                    onHideReplies={onHideRepliesHandler(item.id)}
-                                    onReplyCreate={onReplyCreateHandler(item.id, item.type)}
-                                    onReplyDelete={onReplyDeleteHandler(item.id)}
-                                    onReplyEdit={onReplyUpdateHandler(item.id)}
-                                    onReplySelect={onCommentSelectHandler(item.id)}
-                                    onShowReplies={onShowRepliesHandler(item.id, item.type)}
-                                    repliesTotalCount={item.total_reply_count}
-                                    replies={item.replies}
-                                    translations={translations}
-                                >
-                                    <AnnotationActivity
+                                {hasNewThreadedReplies && onAnnotationSelect ? (
+                                    // TODO: legitimate, pre-existing typing issue that was previously undetected
```

Comment:
So remember hasNewThreadedReplies will go away as a check as will ActivityThread. In this case are we saying that we don't even want to render BaseComment if onAnnotationsSelect is not passed in b/c right now it's an optional in this class. Should we default onSelect to a noop in the AnnotationActivityLinkProvider if nothing is passed in for that prop? If we don't do this then we probably need to make onAnnotationsSelect required otherwise annotations could potentially never be rendered. Lemme know your thoughts.

Reply:
Yeah, that's a good point. This is part of a larger problem with `ActiveState` (the same problem that is responsible for the `$FlowFixMe`s I had to add to this file): we don't know which type of activity item is being passed in. I think what we need to do is create a union for `ActivityFeedProps` (currently, just called `Props`, fyi). Maybe not exactly like this, but here's the idea:
```
type ActivityFeedAnnotationProps = ActivityFeedCommonProps & {
    items: FeedItemsWithAnnotations,
    onAnnotationEdit: OnAnnotationEdit,
    onAnnotationSelect: OnAnnotationSelect,
    onAnnotationStatusChang: OnAnnotationsSelect,
};

type ActivityFeedWithoutAnnotationProps = ActivityFeedCommonProps & {
    items: FeedItemsWithoutAnnotations,
};

type ActivityFeedProps = ActivityFeedAnnotationProps | ActivityFeedWithoutAnnotationProps;
```
This way, if it's possible for `item.type === FEED_ITEM_TYPE_ANNOTATION`, `onAnnotationSelect` (et al) also have to be defined, so we don't have to perform this check (and the types would prevent someone from passing in an annotation item without the annotation functions).

I'm not 100% sure how this would work with Flow. It might require some typeguards (more than the switch) to disambiguate which of the activity feed item types we have. Anyway, we could try it as part of this PR or as a follow up. Jlmk!

Reply:
We decided to break this out into a separate PR, since it's probably going to be somewhat nontrivial.